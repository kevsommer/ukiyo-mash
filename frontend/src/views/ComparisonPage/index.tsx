import { useState, useEffect } from 'react';
import axios from 'axios';
import styled from '@emotion/styled';
import { Container, Paper } from '@mui/material';
import BasicCard from '../../components/layout/BasicCard';

const ContainedImage = styled('img')`
  height: 25vh;
  width: 20vw;
  object-fit: scale-down;
  display: flex;
  margin: 0 auto;
`;

const HorizontalContainer = styled(Container)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 5vh;
`;

const baseURL = 'http://localhost:8000';

interface Artwork {
  id: number;
  object_img_small: string;
  artist_name: string;
  title: string;
  object_begin_date: number;
  object_end_date: number;
}

const ComparisonPage = () => {
  const [leftArtwork, setLeftArtwork] = useState<Artwork | undefined>(
    undefined
  );
  const [rightArtwork, setRightArtwork] = useState<Artwork | undefined>(
    undefined
  );

  const [reload, setReload] = useState<boolean>(true);

  useEffect(() => {
    if (reload) {
      axios.get(`${baseURL}/items/random`).then((res) => {
        setLeftArtwork(res.data[0]);
        setRightArtwork(res.data[1]);
      });
      setReload(false);
    }
  }, [reload]);

  const voteArtwork = (id: number): void => {
    if (leftArtwork?.id === id) {
      axios
        .patch(`${baseURL}/items/vote/${id}/${rightArtwork?.id}`)
        .then((res) => {
          if (res.status === 200) {
            setReload(true);
            console.log(res.data);
          }
        });
    } else {
      axios
        .patch(`${baseURL}/items/vote/${leftArtwork?.id}/${id}`)
        .then((res) => {
          if (res.status === 200) {
            setReload(true);
            console.log(res.data);
          }
        });
    }
  };

  return (
    <HorizontalContainer maxWidth='md'>
      <Paper elevation={3} style={{ width: '45%', height: '25vh' }}>
        <ContainedImage src={leftArtwork?.object_img_small} alt='Loading...' />
        <BasicCard {...leftArtwork} onVoteClick={voteArtwork} />
      </Paper>
      <Paper elevation={3} style={{ width: '45%', height: '25vh' }}>
        <ContainedImage src={rightArtwork?.object_img_small} alt='Loading...' />
        <BasicCard {...rightArtwork} onVoteClick={voteArtwork} />
      </Paper>
    </HorizontalContainer>
  );
};

export default ComparisonPage;
