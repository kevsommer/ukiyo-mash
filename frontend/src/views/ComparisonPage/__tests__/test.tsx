import { render } from '@testing-library/react';
import ComparisonPage from '../index';

test('renders correctly', () => {
  const component = render(<ComparisonPage />);
  expect(component).toMatchSnapshot();
});
