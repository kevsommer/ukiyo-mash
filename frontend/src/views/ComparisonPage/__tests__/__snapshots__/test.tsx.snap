// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1vi5va3-MuiContainer-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
          style="width: 45%; height: 50vh;"
        >
          <img
            alt="..."
            class="css-104w4nu"
          />
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1su4yfi-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-1dgf9yl-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
              />
              <div
                class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-1eav3hy-MuiTypography-root"
              >
                ca. 
                -
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
              style="margin-top: auto; display: flex; flex-direction: row; justify-content: space-between;"
            >
              <div>
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Learn More
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Vote
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FullscreenIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
          style="width: 45%; height: 50vh;"
        >
          <img
            alt="Loading..."
            class="css-104w4nu"
          />
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1su4yfi-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-1dgf9yl-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
              />
              <div
                class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-1eav3hy-MuiTypography-root"
              >
                ca. 
                -
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
              style="margin-top: auto; display: flex; flex-direction: row; justify-content: space-between;"
            >
              <div>
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Learn More
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Vote
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FullscreenIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1vi5va3-MuiContainer-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
        style="width: 45%; height: 50vh;"
      >
        <img
          alt="..."
          class="css-104w4nu"
        />
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1su4yfi-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-1dgf9yl-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            />
            <div
              class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1eav3hy-MuiTypography-root"
            >
              ca. 
              -
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            style="margin-top: auto; display: flex; flex-direction: row; justify-content: space-between;"
          >
            <div>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Learn More
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Vote
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FullscreenIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
        style="width: 45%; height: 50vh;"
      >
        <img
          alt="Loading..."
          class="css-104w4nu"
        />
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1su4yfi-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-1dgf9yl-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            />
            <div
              class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1eav3hy-MuiTypography-root"
            >
              ca. 
              -
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            style="margin-top: auto; display: flex; flex-direction: row; justify-content: space-between;"
          >
            <div>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Learn More
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Vote
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FullscreenIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
